docs/examples/resnet50_trt/README.md

```diff
@@ -0,0 +1,255 @@
+sh setup_resnet50_trt_example.sh
+```
+
+Below we add step-by-step guide, how to set up and run this example.
+If you used the `setup_resnet50_trt_example.sh` script, please skip to
+[Run Triton Inference Server](#run-triton-inference-server) section.
+
+### Step-by-step guide
+
+Prepare directory structure for the model repository.
+
+```
+mkdir -p model_repository/dali/1
+mkdir -p model_repository/ensemble_dali_resnet50/1
+mkdir -p model_repository/resnet50_trt/1
+```
+
+####  Build TensorRT via ONNX
+
+##### 1.  Converting PyTorch Model to ONNX-model 
+
+Run `onnx_exporter.py` for conversion using PyTorch model to ONNX model. In this case, ResNet50 model is converted to ONNX format. `width` and `height` dims are fixed at 224 but dynamic axes arguments for dynamic batch is used. 
+
+```
+docker run -it --gpus=all -v $(pwd):/workspace nvcr.io/nvidia/pytorch:20.12-py3 bash
+python onnx_exporter.py --save model.onnx
+```
+
+##### 2. Building ONNX-model to TensorRT engine
+
+Set the arguments for enabling fp16 precision `--fp16` and for dynamic shapes using a profile `--minShapes`, `--optShapes`, and `maxShapes` with `--explicitBatch`
+
+```python
+trtexec --onnx=model.onnx --saveEngine=./model_repository/resnet50_trt/1/model.plan --explicitBatch --minShapes=input:1x3x224x224 --optShapes=input:1x3x224x224 --maxShapes=input:256x3x224x224 --fp16
+```
+
+##### 3. Serialize DALI pipeline 
+
+Run `serialize_dali_pipeline.py` for generating  DALI pipeline. 
+
+```
+python serialize_dali_pipeline.py --save ./model_repository/dali/1/model.dali
+```
+
+If had a dedicated hardware decoder, the hardware is available. Set the preprocessing pipeline  `resize` and `normalize` using `dali.pipeline.Pipeline` and serialize pipeline using `Pipeline.serialize`
+
+````python
+pipe = dali.pipeline.Pipeline(batch_size=256, num_threads=4, device_id=0)
+with pipe:
+    images = dali.fn.external_source(device="cpu", name="DALI_INPUT_0")
+    images = dali.fn.image_decoder(images, device="mixed", output_type=types.RGB)
+    images = dali.fn.resize(images, resize_x=224, resize_y=224)
+    images = dali.fn.crop_mirror_normalize(images,
+                                           dtype=types.FLOAT,
+                                           output_layout="CHW",
+                                           crop=(224, 224),
+                                           mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
+                                           std=[0.229 * 255, 0.224 * 255, 0.225 * 255])
+
+
+    pipe.set_outputs(images)
+    pipe.serialize(filename=args.save)
+````
+
+
+
+## Run Triton Inference Server
+
+![](./images/ensemble.PNG)
+
+```bash
+model_repository
+├── dali
+│   ├── 1
+│   │   └── model.dali
+│   └── config.pbtxt
+├── ensemble_dali_resnet50
+│   ├── 1
+│   └── config.pbtxt
+└── resnet50_trt
+    ├── 1
+    │   └── model.plan
+    ├── config.pbtxt
+    └── labels.txt
+```
+
+For preprocessing using DALI 
+
+```
```

Comment:
I decided to get rid of the `config.pbtxt` snippets, since it's all in the repo. I only retained the layout of model repo, cause it's something that is generated during the setup.